src/encoded/commands/simulate_environment.py

```
@@ -0,0 +1,54 @@
+import logging
+import os
+import argparse
+import structlog
+from contextlib import contextmanager
+from dcicutils.qa_utils import override_environ
+from dcicutils.env_utils import is_stg_or_prd_env
+from dcicutils.beanstalk_utils import get_beanstalk_environment_variables
+from deploy.generate_production_ini import CGAPIniFileManager
+
+
+logger = structlog.getLogger(__name__)
+EPILOG = __doc__
+
+
+def build_ini_file(environment, use_prod):
+    """ Wrapper method for main functionality that can be invoked directly by others.
+
+        :param environment: environment to simulate
+        :param use_prod: an extra value that must be true to simulate staging/production
+        :returns: True if successful, False otherwise
+    """
+    if is_stg_or_prd_env(environment) and not use_prod:
+        return False
```

To me instead of returning True/False, I'd just raise an error for the False case, since you know at this point how to describe the problem and otherwise the client has to guess.  Even so, this is not a blocker. Just thinking aloud.